integration_tests/ethereum.go

```diff
@@ -0,0 +1,189 @@
+          "type": "uint256"
+        }
+      ],
+      "name": "approve",
+      "outputs": [
+        {
+          "internalType": "bool",
+          "name": "",
+          "type": "bool"
+        }
+      ],
+      "stateMutability": "nonpayable",
+      "type": "function"
+    }
+]
+`
+
+const balanceOfERC20ABIJSON = `
+[
+	{
+      "inputs": [
+        {
+          "internalType": "address",
+          "name": "account",
+          "type": "address"
+        }
+      ],
+      "name": "balanceOf",
+      "outputs": [
+        {
+          "internalType": "uint256",
+          "name": "",
+          "type": "uint256"
+        }
+      ],
+      "stateMutability": "view",
+      "type": "function"
+    }
+]
+`
+
+const allowanceERC20ABIJSON = `
+[
+	{
+      "inputs": [
+        {
+          "internalType": "address",
+          "name": "owner",
+          "type": "address"
+        },
+        {
+          "internalType": "address",
+          "name": "spender",
+          "type": "address"
+        }
+      ],
+      "name": "allowance",
+      "outputs": [
+        {
+          "internalType": "uint256",
+          "name": "",
+          "type": "uint256"
+        }
+      ],
+      "stateMutability": "view",
+      "type": "function"
+    }
+]
+`
+
+func packCall(abiString, method string, args []interface{}) []byte {
+	encodedCall, err := abi.JSON(strings.NewReader(abiString))
+	if err != nil {
+		panic(sdkerrors.Wrap(err, "bad ABI definition in code"))
+	}
+	abiEncodedCall, err := encodedCall.Pack(method, args...)
+	if err != nil {
+		panic(sdkerrors.Wrap(err, "error packing calling"))
+	}
+	return abiEncodedCall
+}
+
+func PackDeployERC20(denom string, name string, symbol string, decimals uint8) []byte {
+	return packCall(gravitytypes.DeployERC20ABIJSON, "deployERC20", []interface{}{
+		denom,
+		name,
+		symbol,
+		decimals,
+	})
+}
+
+func PackSendToCosmos(tokenContract common.Address, destination sdk.AccAddress, amount sdk.Int) []byte {
+	destinationBytes, _ := byteArrayToFixByteArray(destination.Bytes())
+	return packCall(gravitytypes.SendToCosmosABIJSON, "sendToCosmos", []interface{}{
+		tokenContract,
+		destinationBytes,
+		amount.BigInt(),
+	})
+}
+
+func UInt256Max() *big.Int {
+	return new(big.Int).Sub(new(big.Int).Lsh(common.Big1, 256), common.Big1)
+}
+
+func PackApproveERC20(spender common.Address) []byte {
+	return packCall(approveERC20ABIJSON, "approve", []interface{}{
+		spender,
+		UInt256Max(),
+	})
+}
+
+func PackBalanceOf(account common.Address) []byte {
+	return packCall(balanceOfERC20ABIJSON, "balanceOf", []interface{}{
+		account,
+	})
+}
+
+func PackAllowance(owner common.Address, spender common.Address) []byte {
+	return packCall(allowanceERC20ABIJSON, "allowance", []interface{}{
+		owner,
+		spender,
+	})
+}
+
+func UnpackEthUInt(bz []byte) sdk.Int {
+	output := big.NewInt(0)
+	output.SetBytes(bz)
+
+	return sdk.NewIntFromBigInt(output)
+}
+
+func byteArrayToFixByteArray(b []byte) (out [32]byte, err error) {
```

Comment:
Just noticed while looking at this that this name is a little dangerous to be left generic, as it can only copy 20 bytes from the input slice.